src/feature/hs/hs_service.h

```diff
@@ -220,6 +220,9 @@ typedef struct hs_service_config_t {
   /* Is this service ephemeral? */
   unsigned int is_ephemeral : 1;
 
+  /* Have we explicitly set HiddenServiceVersion? */
+  unsigned int hs_version_explicitly_set : 1;
+
```

Comment:
Super nitpick here: Can you put that next to the `version` so we have the parameters from the same context next to each other?